wdl-lint/src/rules/whitespace.rs

```diff
@@ -81,6 +81,18 @@ impl Rule for WhitespaceRule {
 impl Visitor for WhitespaceRule {
     type State = Diagnostics;
 
+    fn comment(&mut self, state: &mut Self::State, comment: &wdl_ast::Comment) {
+        let comment_str = comment.as_str();
+        if comment_str.ends_with(|char: char| char.is_whitespace()) {
+            // Trailing whitespace
+            state.exceptable_add(
+                trailing_whitespace(comment.span()),
```

Comment:
The `Span` passed into `trailing_whitespace()` gets highlighted (or underlined with a squiggle). Here, you are highlighting the entire comment in the diagnostic with `comment.span()`.

It would be preferable if the diagnostic only highlighted the trailing whitespace (and not just the last character. It should should look backwards and highlight any continuous whitespace at the end of the string).

Can you make that change? After you do that, you should update the test you added with a variety of trailing whitespace to make sure it all gets highlighted correctly.

Reply:
Thanks for the explanation, I'll get that fixed up after my classes today :+1:

Reply:
@a-frantz - The printed diagnostic (which predates this PR) is `remove the trailing whitespace from this line`. Should that be `remove this trailing whitespace`, since we're highlighting just the excess whitespace in our diagnostics?

Reply:
That's a good catch, I think you're right. @BradenEverson this is a nit over language that predates your PR, but should be fixed while we're working on the file. [Line we're discussing](https://github.com/stjude-rust-labs/wdl/blob/main/wdl-lint/src/rules/whitespace.rs#L37). Please make the change to language that @adthrasher pointed out.

Reply:
@adthrasher similar thought of just cleaning up while we're here: should `trailing whitespace` really be at a `warning` level? That feels more like `note` territory. Not sure how important that distinction is.

Reply:
After spamming @a-frantz's Slack, I agree that `note` is more appropriate for this lint.

Reply:
That probably applies to all three diagnostics in this rule.

Reply:
The span has been updated to only highlight the trailing whitespace and I updated the language to what you said @adthrasher. Do we want to also do the change from warning to note for all Whitespace diagnostics as a part of this PR?

Reply:
> The span has been updated to only highlight the trailing whitespace and I updated the language to what you said @adthrasher. Do we want to also do the change from warning to note for all Whitespace diagnostics as a part of this PR?

I pushed a commit changing all Whitespace diagnostics to notes, if this isn't the way we want to move with this we can revert the commit.